server/game/game.js

```diff
@@ -814,6 +814,7 @@ class Game extends EventEmitter {
     }
 
     continue() {
+        this.effectEngine.reapplyStateDependentEffects();
```

Comment:
So, this one is interesting, when nested steps call continue themselves, effects are not recalculated.

e.g. Core Cat vs Tyrion

Challenge is initiated, stealth targets are chosen, 'onChallenge' event is raised before the game effects are recalculated

Not sure if there's a better way to fix that than this